lib/auth/google_oauth2_authenticator.rb

```diff
@@ -35,8 +36,25 @@ def register_middleware(omniauth)
         # the JWT can fail due to clock skew, so let's skip it completely.
         # https://github.com/zquestz/omniauth-google-oauth2/pull/392
         strategy.options[:skip_jwt] = true
+        strategy.options[:request_groups] = provides_groups?
+
+        if provides_groups?
+          strategy.options[:scope] = "#{strategy_class::DEFAULT_SCOPE},#{strategy_class::GROUPS_SCOPE}"
+        end
       }
     }
-    omniauth.provider :google_oauth2, options
+    omniauth.provider strategy_class, options
+  end
+
+  def after_authenticate(auth_token, existing_account: nil)
+    result = super
+    if provides_groups? && (groups = auth_token[:extra][:raw_groups]).any?
```

Comment:
We should process an empty array, otherwise a user will never be 'removed' from their final group (see spec addition)
```suggestion
    if provides_groups? && (groups = auth_token[:extra][:raw_groups])
```